server/metadata/tenant.go

```diff
@@ -1251,40 +1250,46 @@ func (tenant *Tenant) createCollection(ctx context.Context, tx transaction.Tx, d
 	return nil
 }
 
-func (tenant *Tenant) updateCollection(ctx context.Context, tx transaction.Tx, database *Database, c *collectionHolder, schFactory *schema.Factory) error {
+func (tenant *Tenant) updateCollection(ctx context.Context, tx transaction.Tx, database *Database, cHolder *collectionHolder, schFactory *schema.Factory) error {
 	var err error
 
-	for _, idx := range schFactory.Indexes.GetIndexes() {
-		meta, ok := c.idxMeta[idx.Name]
-		if !ok {
-			// these are the new indexes present in the new collection
-			meta, err = tenant.metaStore.CreateIndex(ctx, tx, idx.Name, tenant.namespace.Id(), database.id, c.id)
-			if err != nil {
-				return err
-			}
-			c.addIndex(idx.Name, meta)
+	primaryKeyIndex := schFactory.PrimaryKey
+	meta, ok := cHolder.primaryIdxMeta[primaryKeyIndex.Name]
+	if !ok {
+		// these are the new indexes present in the new collection
+		meta, err = tenant.metaStore.CreatePrimaryIndex(ctx, tx, primaryKeyIndex.Name, tenant.namespace.Id(), database.id, cHolder.id)
```

Comment:
this should not happen now as it is backward incompatible change, let's return a bad request error that updating primary key is not allowed.

Reply:
Something like:
```
if !ok {
		return errors.InvalidArgument("cannot create a new primary key name")
	}
```